kld/src/cli/main.rs

```diff
@@ -118,37 +126,64 @@ fn main() {
 }
 
 fn run_command(args: Args) -> Result<()> {
-    let api = Api::new(&args.target, &args.cert_path, &args.macaroon_path)?;
-
     let output = match args.command {
-        Command::GetInfo => api.get_info()?,
-        Command::GetBalance => api.get_balance()?,
-        Command::NewAddress => api.new_address()?,
-        Command::Withdraw {
-            address,
-            satoshis,
-            fee_rate,
-        } => api.withdraw(address, satoshis, fee_rate)?,
-        Command::ListChannels => api.list_channels()?,
-        Command::ListPeers => api.list_peers()?,
-        Command::ConnectPeer { public_key } => api.connect_peer(public_key)?,
-        Command::DisconnectPeer { public_key } => api.disconnect_peer(public_key)?,
-        Command::OpenChannel {
-            public_key,
-            sats: satoshis,
-            push_msat,
-            announce,
-            fee_rate,
-        } => api.open_channel(public_key, satoshis, push_msat, announce, fee_rate)?,
-        Command::SetChannelFee {
-            id,
-            base_fee,
-            ppm_fee,
-        } => api.set_channel_fee(id, base_fee, ppm_fee)?,
-        Command::CloseChannel { id } => api.close_channel(id)?,
-        Command::NetworkNodes { id } => api.list_network_nodes(id)?,
-        Command::NetworkChannels { id } => api.list_network_channels(id)?,
+        Command::GetInfo
+        | Command::GetBalance
+        | Command::NewAddress
+        | Command::Withdraw { .. }
+        | Command::ListChannels
+        | Command::ListPeers
+        | Command::ConnectPeer { .. }
+        | Command::DisconnectPeer { .. }
+        | Command::OpenChannel { .. }
+        | Command::SetChannelFee { .. }
+        | Command::CloseChannel { .. }
+        | Command::NetworkNodes { .. }
+        | Command::NetworkChannels { .. } => {
+            if let (Some(target), Some(cert_path), Some(macaroon_path)) =
```

Comment:
I get that these are not required for SystemInfo command, but they will always be set in our deployments. So I think its best to keep it simple and not make them optional args.

Reply:
If that, what should we pass for `cert_path` and `macaroon_path` here

https://github.com/kuutamolabs/lightning-knd/blob/512006a1c79e4ecfac50807561e9767cea1859c3/mgr/src/nixos_rebuild.rs#L85-L87

And the code in nixos_rebuild.rs will not be simple, and we need to pass not really used parameters `cert_path` and `macaroon_path` into the function.

Reply:
Ok fair enough.

Reply:
Actually these should remain required args. The manager is calling kld-cli via SSH so it is using the wrapper script that sets the cert_path and macaroon_path to the readonly macaroon.

Reply:
Are `cert_path` and `macaroon_path` meaningful?
As I known, `cert_path` is `/var/lib/kld/certs/ca.pem`, and `macaroon_path` is `/var/lib/kld/macaroons/admin.macaroon`.  These just are paths, event if the content change the path are still the same.